test/e2e/aws_util_test.go

```diff
@@ -0,0 +1,196 @@
+	input := &ec2.DescribeInternetGatewaysInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("attachment.vpc-id"),
+				Values: []*string{aws.String(vpcID)},
+			},
+		},
+	}
+
+	resultIG, err := svc.DescribeInternetGateways(input)
+	if err != nil {
+		return nil, fmt.Errorf("Error describing Internet Gateways: %v", err)
+	}
+
+	// Extract Internet Gateway IDs
+	//igws := make([]string, 0)
+	var igws []*ec2.InternetGateway
+	for _, igw := range resultIG.InternetGateways {
+		//igws = append(igws, aws.StringValue(igw.InternetGatewayId))
+		igws = append(igws, igw)
+	}
+
+	// Find public subnets associated with the Internet Gateways
+	publicSubnets := make(map[string]struct{})
+	for _, igw := range igws {
+		input := &ec2.DescribeRouteTablesInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("route.gateway-id"),
+					//Values: []*string{aws.String(igw)},
+					Values: []*string{igw.InternetGatewayId},
+				},
+			},
+		}
+
+		result, err := svc.DescribeRouteTables(input)
+		if err != nil {
+			return nil, fmt.Errorf("Error describing Route Tables: %v", err)
+		}
+
+		for _, rt := range result.RouteTables {
+			for _, assoc := range rt.Associations {
+				if assoc != nil && assoc.SubnetId != nil {
+					publicSubnets[aws.StringValue(assoc.SubnetId)] = struct{}{}
+				}
+			}
+		}
+	}
+	return publicSubnets, nil
+}
+
+// createSession creates a session using decoded AWS credentials.
+func createSession(region, decodedAccessKeyID, decodedSecretAccessKey string) (*session.Session, error) {
+	return session.NewSession(&aws.Config{
+		Region:      aws.String(region),
+		Credentials: credentials.NewStaticCredentials(decodedAccessKeyID, decodedSecretAccessKey, ""),
+	})
+}
+
+// getVPCID fetched AWS VPC ID tagged with cluster name as tag key.
+/*func getVPCID(clusterName string, svc *ec2.EC2) (string, error) {
+	// Define the filter to search for instances with the specified tag
+	tagKey := "kubernetes.io/cluster/" + clusterName
+	tagValue := "owned"
+	filters := []*ec2.Filter{
+		{
+			Name:   aws.String("tag:" + tagKey),
+			Values: []*string{aws.String(tagValue)},
+		},
+	}
+
+	// Describe instances that match the filter
+	result, err := svc.DescribeInstances(&ec2.DescribeInstancesInput{
+		Filters: filters,
+	})
+	if err != nil {
+		return "", fmt.Errorf("Error describing instances: %v", err)
+	}
+
+	// Check if any instances were found
+	if len(result.Reservations) == 0 || len(result.Reservations[0].Instances) == 0 {
+		return "", fmt.Errorf("No instances found with the specified tag, or unable to determine VPC ID.")
+	}
+	instances := result.Reservations[0].Instances
+	for _, instance := range instances {
+		if instance.VpcId != nil {
+			// Extract and return the VPC ID
+			return aws.StringValue(instance.VpcId), nil
+		}
+	}
+	return "", fmt.Errorf("Can't find VPC ID due to error: %v", err)
+}
+*/
+
+// GetVPCId return the VPC ID of the cluster
+func GetVPCId(ec2Client *ec2.EC2, clusterName string) (string, error) {
```

Comment:
This function is not used outside of this package:
```suggestion
func getVPCId(ec2Client *ec2.EC2, clusterName string) (string, error) {
```